### PR TITLE
Add valid preferred_ext check in wp_mime_type_icon and unit testcase …

### PR DESCRIPTION
**Pull request by itzmekhokan**
…for it

Add `$preferred_ext` check in function `wp_mime_type_icon` with proper unit test cases.

Trac ticket: [#60610](https://core.trac.wordpress.org/ticket/60610)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/post.php [hunk 1]
```diff
@@ -6862,6 +6862,11 @@ function wp_mime_type_icon( $mime = 0, $preferred_ext = '.png' ) {
 		$icon = wp_cache_get( "mime_type_icon_$mime" );
 	}
 
+	// Check preferred file format is available and if it is begins with a period or not.
```
Comment (joedolson) [inline src/wp-includes/post.php]: I think this is a better comment:

// Check if preferred file format variable is present and is a valid file extension.

File: tests/phpunit/tests/post/attachments.php [hunk 2]
```diff
@@ -519,4 +519,15 @@ public function test_wp_mime_type_icon_video() {
...
+	/**
+	 * @ticket 60610
+	 */
+	public function test_wp_mime_type_icon_video_with_preferred_ext() {
+		$icon1 = wp_mime_type_icon( 'video/mp4', '.png' ); // Added `$preferred_ext` parameter.
+		$icon2 = wp_mime_type_icon( 'video/mp4', 'png' ); // Added `$preferred_ext` parameter without period.
+
+		$this->assertStringContainsString( 'images/media/video.png', $icon1 );
+		$this->assertStringContainsString( 'images/media/video.png', $icon2 );
+	}
```
Comment (joedolson) [inline tests/phpunit/tests/post/attachments.php:532]: For both assertions, include the description argument so failed tests will include context. E.g.

`$this->assertStringContainsString( 'images/media/video.png', $icon1, 'Mime type icon correctly returned with ".png" argument.' );`

File: src/wp-includes/post.php (hunk 1 above)
Comment (joedolson) [reply src/wp-includes/post.php]: Maybe more like "validly formatted file extension", since we're not actually checking the extension validity.

**Review by joedolson — CHANGES_REQUESTED** [review]
Minor changes requested; but this looks good, overall.

File: tests/phpunit/tests/post/attachments.php (hunk 2 above)
Comment (itzmekhokan) [reply tests/phpunit/tests/post/attachments.php:532]: @joedolson I've made the suggested changes.

Comment (joedolson) [reply tests/phpunit/tests/post/attachments.php:532]: The second context string should be "returned with "png" argument'', but I'll fix that in the commit.

**Review by joedolson — APPROVED** [review]
One minor change that I'll make at commit.

**Comment by joedolson** [comment]
In r58248

**Pull request closed without merging**